File: Easy/168. Excel Sheet Column Title/168. Excel Sheet Column Title.java
class Solution {
    public String convertToTitle(int columnNumber) {
        StringBuilder sb = new StringBuilder();
        
        while (columnNumber > 0) {
            columnNumber--; // Adjust to 0-based index
            int remainder = columnNumber % 26;
            char digit = (char) ('A' + remainder);
            sb.insert(0, digit);
            columnNumber /= 26;
        }
        
        return sb.toString();
    }
}

File: Easy/168. Excel Sheet Column Title/168. Excel Sheet Column Title.rs
impl Solution {
    pub fn convert_to_title(column_number: i32) -> String {
        let mut result = String::new();
        let mut column_number = column_number;

        while column_number > 0 {
            column_number -= 1; // Adjust to 0-based index
            let remainder = column_number % 26;
            let digit = (b'A' + remainder as u8) as char;
            result.insert(0, digit);
            column_number /= 26;
        }

        result
    }
}
